Tests/Integration.Tests.ps1

```diff
@@ -0,0 +1,97 @@
+
+        BeforeAll {
+            $testData = Join-Path (Split-Path -Parent $PSScriptRoot) "Test-Data"
+            $htpasswdPath = Join-Path $testData 'DockerRegistry'
+            $dockerImages = Join-Path $testData 'DockerImage'
+            $removeImageCommand = 'docker image rm --force localhost:5000/integration-testcase-2:latest'
+            $pruneImageCommand = 'docker system prune --force'
+            $startRegistryCommand = "docker run -d -p 5000:5000 --name registry -v ${htpasswdPath}:/auth -e 'REGISTRY_AUTH=htpasswd' -e 'REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm' -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd registry:2"
+            Invoke-Command $removeImageCommand
+            Invoke-Command $pruneImageCommand
+            Invoke-Command $startRegistryCommand
+        }
+
+        AfterAll {
+            Invoke-Command 'docker container stop registry'
+            Invoke-Command 'docker container rm -v registry'
+        }
+
+        BeforeEach {
+            $script:backupDebugPref = $DebugPreference
+            $DebugPreference='Continue'
+            $script:backupLocation = Get-Location
+        }
+
+        AfterEach {
+            $DebugPreference=$script:DebugPreference
+            Set-Location $script:backupLocation
+        }
+
+        It "Use case #1: can derive docker image name and tag in one go" {
+            $exampleReposPath = Join-Path $testData "ExampleRepos"
+            $location = Join-Path $exampleReposPath "3.0/servercore/amd64"
+            Set-Location $location
+            New-FakeGitRepository $location
+
+            $result = Find-ImageName $location
+
+            $result = Format-DockerTag | Invoke-DockerBuild -ImageName $result.ImageName
+
+            $result.Dockerfile | Should -BeLike "*Dockerfile"
+            $result.ImageName | Should -Be "dockerbuild-pwsh"
+            $result.Tag | Should -Be  "3.0-servercore-amd64"
+        }
+
+        It 'supports docker login' {
+            $result = Invoke-DockerLogin -Username 'admin' -Password (ConvertTo-SecureString 'password' –asplaintext –force) -Registry 'localhost:5000'
+            $result.ExitCode | Should -Be 0
+        }
+
+        It "Use case #2: can build and push in one go" {
+
+            $exampleReposPath = Join-Path $testData "ExampleRepos"
+            $location = Join-Path $exampleReposPath "3.0/servercore/amd64"
+            Set-Location $location
+            New-FakeGitRepository $location
+
+            Invoke-DockerLogin -Username 'admin' -Password (ConvertTo-SecureString 'password' –asplaintext –force) -Registry 'localhost:5000'
+            Invoke-DockerBuild -ImageName 'integration-testcase-2' -Registry 'localhost:5000' | Invoke-DockerPush -Registry 'localhost:5000'
+
+            $result = Invoke-DockerPull -Registry 'localhost:5000' -ImageName 'integration-testcase-2' -Tag 'latest'
+            Write-Debug $result.Output
+            $result.ExitCode | Should -Be 0
+        }
+
+        # It 'can login' {
```

Comment:
This test is implemented right above? why not delete commented code?

Reply:
See updated code base. 😄